### PR TITLE
Add explicit support for certain objectives and apply the sigmoid when appropriate

### PR DESCRIPTION
**Pull request by ndawe**
This also fixes a lingering base_score issue. This hasn't been a problem with logistic objectives when base_score is 0.5 as that reduces to a baseline raw margin of 0. But if base_score is anything else, or if the objective is something like `reg:squarederror` then xgb2code generated code will produce incorrect output.

## Summary by CodeRabbit

* **Tests**
  * Expanded automated test coverage with a new case validating generation and execution for the "reg-squarederror" model. This exercises the end-to-end code-generation and run workflow for an additional model, increasing confidence in model handling and reducing regression risk.

### CODE REVIEW

File: gen/testdata/reg-squarederror/generate-model.py [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+data = sklearn.datasets.load_diabetes(return_X_y=False)
+df = pd.DataFrame(data.data)  # pylint: disable=no-member
+# make some columns NaN at random.
+df = df.mask(np.random.random(df.shape) < 0.2)
+
+X_train, X_test, y_train, y_test = train_test_split(
+    df,
+    data.target,  # pylint: disable=no-member
+    test_size=0.30,
+)
```
[the remark this answers is not in the document]
Comment (ndawe) [reply gen/testdata/reg-squarederror/generate-model.py:24]: The script already sets the global numpy random seed, which is sufficient. This also follows the approach in `gen/testdata/breast-cancer/generate-model.py`

File: gen/parse.go [hunk 2]
```diff
@@ -114,3 +140,52 @@ func parseTreeInfo(xt xgbTree) (*node, error) {
...
+		)
+	}
+
+	baseScore, err := parseBaseScore(x.Learner.LearnerModelParam.BaseScore)
+	if err != nil {
+		return modelMeta{}, fmt.Errorf(
+			"error parsing base_score: %w",
+			err,
+		)
+	}
```
[the remark this answers is not in the document]
Comment (ndawe) [reply gen/parse.go:238]: Added this in parseBaseScore directly.

File: gen/parse_test.go [hunk 3]
```diff
@@ -127,6 +128,106 @@ func TestParseTreeInfo(t *testing.T) {
...
+			baseScore:     "0",
+			expectError:   true,
+		},
+		{
+			name:          "logistic with base_score 1",
+			objectiveName: "binary:logistic",
+			baseScore:     "1",
+			expectError:   true,
+		},
+	}
```
[the remark this answers is not in the document]
Comment (ndawe) [reply gen/parse_test.go:323]: Done

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,9 +1,17 @@
 # CHANGELOG
 
-## 1.0.0
+## 1.0.0 (2026-06-12)
```
Comment (horgh) [inline CHANGELOG.md]: I assume we'll need to update the date, sorry about that. If you're not planning to release immediately, could we leave a date off as well? Our convention is to only put a date when we're actually doing the release, which helps when updating the changelog if we have additional changes over time without releasing. Maybe in this case we'll be releasing now/soon so it won't be a concern though.

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,9 +1,17 @@
...
 - Refactored the codebase to allow using the code generation functionality as a
   library.
+- Added support for the `reg:logistic`, `binary:logitraw`, `reg:squarederror`,
+  `reg:linear`, `reg:absoluteerror`, `reg:pseudohubererror`, and
+  `reg:quantileerror` objectives. The model's `base_score` is now applied as an
+  intercept, and the sigmoid is only applied for the logistic objectives
+  (`binary:logistic` and `reg:logistic`). Unsupported objectives, multi-output
+  models (such as multi-target regression or multi-quantile
+  `reg:quantileerror`), and forest models (`num_parallel_tree` greater than one)
+  trained with early stopping now produce a clear error.
```
Comment (horgh) [inline CHANGELOG.md:14]: Should we update the README with any of this? The Model Support section for example. I think it is now wrong.

**Review by horgh — CHANGES_REQUESTED** [review]
Looks great! Just a couple minor comments.

File: CHANGELOG.md (hunk 5 above)
Comment (ndawe) [reply CHANGELOG.md:14]: Yes, thanks. I updated the README in a new commit. I can squash that before merge.

File: CHANGELOG.md (hunk 4 above)
Comment (ndawe) [reply CHANGELOG.md]: I removed the date.

**Review by horgh — APPROVED** [review]
